pyomo/gdp/plugins/bigm.py

```diff
@@ -421,34 +387,109 @@ def _bigM_relax_disjunct(self, obj, transBlock, bigM, suffix_list):
         # comparing the two relaxations.
         #
         # Transform each component within this disjunct
-        self._transform_block_components(obj, obj, infodict, bigM, suffix_list)
+        self._transform_block_components(obj, obj, bigM, suffix_list)
 
-        # deactivate disjunct so we know we've relaxed it
+        # deactivate disjunct to keep the writers happy
         obj._deactivate_without_fixing_indicator()
-        infodict['relaxed'] = True
 
-    def _transform_block_components(self, block, disjunct, infodict,
-                                    bigM, suffix_list):
-        # Look through the component map of block and transform
+    def _transform_block_components(self, block, disjunct, bigM, suffix_list):
+        # We first need to find any transformed disjunctions that might be here
+        # because we need to move their transformation blocks up onto the parent
+        # block before we transform anything else on this block 
```

Comment:
I think I must be missing something in the rationale for why we are moving these blocks up?

Reply:
Because for nested disjunctions, it is sufficient to relax only the XOR constraint of the inner disjunction, so after transforming the inner disjunction, we move all the transformed constraints from its disjuncts up to avoid transforming them again.

Reply:
Ah, gotcha. So that you don't end up with `g(x) <= M(1-y1) + M(1-y2)`.